### ISSUE TITLE
MissingField support per field

### ISSUE THREAD

**Issue by darren-v-excelian** [JoshClose/CsvHelper#556]
Currently MissingField only supports 'all or nothing'. It would be useful if it could perform per column validation? e.g.

Map( m => m.Id ).Name( "The Id Column" ).NotNull();

or perhaps go further and add support for a custom validator?

Map( m => m.Id ).Name( "The Id Column" ).WithValidator((context)=> myvalidator(context) );

**Comment by JoshClose** [comment]
What happens if it fails validation? Throws an exception?

**Comment by darren-v-excelian** [comment]
For the .NotNull() then throw CsvMissingFieldException
For the custom validator...perhaps let validator implementation decide. It could log or throw...

**Comment by JoshClose** [comment]
note: #479 should be done before this.

**Comment by JoshClose** [comment]
There will be a method `Validate`. Here's how it can be used.

```cs
Map( m => m.Id ).Validate( string field => !string.IsNullOrEmpty( field ) );
```

If validation fails, a `CsvValidationException` will be thrown.

If you don't want the exception thrown and want to log instead, you can just log, then return true. If you want to throw a custom exception, just throw one in this method. Keep in mind all exception are wrapped with some sort of `CsvHelperException` with extra data on it.

**Comment by JoshClose** [comment]
Done. 03c07cd0aac234b6c4d2cc73a3fd008b044c140c

**Comment by darren-v-excelian** [comment]
Thanks for this commit.

**Comment by ruifrvaz** [comment]
Map( m => m.Id ).Validate( string field => !string.IsNullOrEmpty( field ) );

was this ever implemented?  I'm using version 27.0.4 and it does not work

**Comment by UnnecessaryPunctuation** [comment]
> Map( m => m.Id ).Validate( string field => !string.IsNullOrEmpty( field ) );
> was this ever implemented? I'm using version 27.0.4 and it does not work
@ruifrvaz , based on unit test :
https://github.com/JoshClose/CsvHelper/blob/dfd643d1b57a83b7ea5099e92e38f87723baac2c/tests/CsvHelper.Tests/Reading/ValidateTests.cs

It's `Map(m => m.Id).Validate(args => !string.IsNullOrEmpty(args.Field));`
